### PR TITLE
Docs: fixed broken links

### LINKED ISSUE #2866: Docs: fix broken links
## Describe the bug
The docs currently have some broken links, see below (some are false positives)

<html>
<body>

Error | URL | Anchor Text | Linked From
-- | -- | -- | --
500 Internal Server Error | https://gist-it.appspot.com/https://github.com/getporter/porter/blob/main/pkg/porter/examples/install_example_test.go | script/src | https://getporter.org/references/library/
500 Internal Server Error | https://gist-it.appspot.com/https://github.com/getporter/porter/blob/main/pkg/porter/examples/capture_output_example_test.go | script/src | https://getporter.org/references/library/
404 Not Found | https://release-v1.porter.sh/reference/file-formats/ | File Formats | https://getporter.org/getting-started/migrate-v0-v1/
404 Not Found | https://github.com/getporter/proposals/blob/main/pep/003-dependency-namespaces-and-labels.md | PEP003 - Advanced Dependencies | https://getporter.org/configuration/configuration/
404 Not Found | https://github.com/getporter/porter/releases/download/learning/ | Learning [302 from https://getporter.org/learning] | https://getporter.org/
404 Not Found | https://release-v1.porter.sh/plugins/kubernetes/ | kubernetes.secrets plugin | https://getporter.org/operator/glossary/
404 Not Found | https://github.com/getporter/porter/releases/download/best-practices/kind/ | KinD [302 from https://getporter.org/best-practices/kind/] | https://getporter.org/operator/quickstart/
404 Not Found | https://github.com/getporter/porter/releases/download/best-practices/minikube/ | Minikube [302 from https://getporter.org/best-practices/minikube/] | https://getporter.org/operator/quickstart/
404 Not Found | https://github.com/getporter/porter/releases/download/operator/administrators/configure-porter-agent.md | Configure the Porter Agent [302 from https://getporter.org/operator/administrators/configure-porter-agent.md] | https://getporter.org/operator/install/
404 Not Found | https://github.com/getporter/porter/releases/download/reference/file-formats/ | Porter Installation document [302 from https://getporter.org/reference/file-formats/] | https://getporter.org/operator/file-formats/
-1 Invalid URL | http://localhost:1313/docs | http://localhost:1313/docs | https://getporter.org/contribute/guide/
500 Internal Server Error | https://gist-it.appspot.com/https://github.com/getporter/porter/blob/main/examples/porter-agent-manifest.yaml | script/src | https://getporter.org/references/docker-images/agent/
500 Internal Server Error | https://gist-it.appspot.com/https://github.com/getporter/examples/blob/main/docker/porter.yaml | script/src | https://getporter.org/references/examples/docker/
404 Not Found | https://github.com/getporter/porter/releases/download/bundle/create/ | Create a Bundle [302 from https://getporter.org/bundle/create/] | https://getporter.org/architecture/
404 Not Found | https://github.com/getporter/porter/releases/download/administrators/airgap/ | move a bundle into a disconnected or airgapped environment [302 from https://getporter.org/administrators/airgap/] | https://getporter.org/architecture/
404 Not Found | https://github.com/getporter/porter/releases/download/distribute-bundles/ | Distribute Bundles [302 from https://getporter.org/distribute-bundles/] | https://getporter.org/architecture/
404 Not Found | https://github.com/getporter/porter/releases/download/img/boat.png | style: #boat [302 from https://getporter.org/img/boat.png] | https://getporter.org/sass/styles.min.css
404 Not Found | https://github.com/getporter/porter/releases/download/examples/hello/ | hello example bundle [302 from https://getporter.org/examples/hello/] | https://getporter.org/getting-started/create-bundle/
404 Not Found | https://github.com/getporter/porter/releases/download/compatible-registries/ | popular registries have been tested with Porter [302 from https://getporter.org/compatible-registries] | https://getporter.org/quickstart/quickstart/
404 Not Found | https://github.com/getporter/porter/releases/download/cli/porter_mixin_install/ | porter mixin install [302 from https://getporter.org/cli/porter_mixin_install/] | https://getporter.org/getting-started/create-bundle/
404 Not Found | https://github.com/getporter/porter/releases/download/administrators/diagnostics/ | write a logfile for each command [302 from https://getporter.org/administrators/diagnostics/] | https://getporter.org/configuration/configuration/
404 Not Found | https://github.com/getporter/porter/releases/download/plugins/azure-keyvault | azure-keyvault [302 from https://getporter.org/plugins/azure-keyvault] | https://getporter.org/operations/create-porter-config/
404 Not Found | https://github.com/getporter/porter/releases/download/inspect-bundles/ | inspect [302 from https://getporter.org/inspect-bundles] | https://getporter.org/operations/examine-bundles/
404 Not Found | https://github.com/getporter/porter/releases/download/plugins/hashicorp-vault | hashicorp-vault [302 from https://getporter.org/plugins/hashicorp-vault] | https://getporter.org/operations/create-porter-config/
404 Not Found | https://github.com/getporter/porter/releases/download/credentials/ | credential sets [302 from https://getporter.org/credentials/] | https://getporter.org/security-features/
404 Not Found | https://github.com/getporter/porter/releases/download/examples/airgap/ | airgap example [302 from https://getporter.org/examples/airgap/] | https://getporter.org/security-features/
404 Not Found | https://github.com/getporter/porter/releases/download/end-users/configuration/ | Configure Porter [302 from https://getporter.org/end-users/configuration/] | https://getporter.org/operations/connect-to-docker/
404 Not Found | https://github.com/getporter/porter/releases/download/end-users/connect-registry/ | Connect to a Docker Registry [302 from https://getporter.org/end-users/connect-registry/] | https://getporter.org/operations/connect-to-docker/
404 Not Found | https://github.com/getporter/porter/releases/download/end-users/connect-docker/ | Connect to Docker [302 from https://getporter.org/end-users/connect-docker/] | https://getporter.org/operations/connect-to-registry/
404 Not Found | https://github.com/getporter/porter/releases/download/administrators/jaeger-trace-example.png | img/src [302 from https://getporter.org/administrators/jaeger-trace-example.png] | https://getporter.org/administration/collect-diag-porter/
404 Not Found | https://github.com/getporter/porter/blob/main/mixin.mk | Makefile [301 from https://getporter.org/src/mixin.mk] | https://getporter.org/development/dist-a-mixin/
404 Not Found | https://github.com/getporter/porter/releases/download/operators/logs/ | saves the logs [302 from https://getporter.org/operators/logs/] | https://getporter.org/troubleshooting/troubleshooting/
404 Not Found | https://github.com/getporter/porter/releases/download/parameters/ | Parameters Overview [302 from https://getporter.org/parameters/] | https://getporter.org/best-practices/credentials-or-parameters/
404 Not Found | https://github.com/getporter/porter/releases/download/cli/ | https://getporter.org/cli [302 from https://getporter.org/cli] | https://getporter.org/contribute/guide/
404 Not Found | https://github.com/getporter/porter/releases/download/reference/file-formats | file formats [302 from https://getporter.org/reference/file-formats] | https://getporter.org/contribute/guide/
404 Not Found | https://github.com/getporter/porter/releases/download/examine-bundles/ | explain [302 from https://getporter.org/examine-bundles] | https://getporter.org/administration/inspecting-bundles/
404 Not Found | https://github.com/getporter/porter/releases/download/docker-images/client/ | porter client [302 from https://getporter.org/docker-images/client/] | https://getporter.org/references/docker-images/agent/
404 Not Found | https://github.com/getporter/porter/releases/download/project/version-strategy/ | version strategy [302 from https://getporter.org/project/version-strategy/] | https://getporter.org/contribute/guide/
404 Not Found | https://github.com/getporter/porter/releases/download/schema/v1/credential-set.schema.json | json schema [302 from https://getporter.org/schema/v1/credential-set.schema.json] | https://getporter.org/references/file-formats/credential-set/
404 Not Found | https://github.com/getporter/porter/releases/download/schema/v1/plugins.schema.json | json schema [302 from https://getporter.org/schema/v1/plugins.schema.json] | https://getporter.org/references/file-formats/plugins/
404 Not Found | https://github.com/getporter/porter/releases/download/schema/v1/parameter-set.schema.json | json schema [302 from https://getporter.org/schema/v1/parameter-set.schema.json] | https://getporter.org/references/file-formats/parameter-set/
404 Not Found | https://github.com/getporter/porter/releases/download/schema/v1/installation.schema.json | json schema [302 from https://getporter.org/schema/v1/installation.schema.json] | https://getporter.org/references/file-formats/installation/
404 Not Found | https://github.com/getporter/porter/releases/download/integrations/credentials | Credential Set [302 from https://getporter.org/integrations/credentials] | https://getporter.org/integrations/kind/
404 Not Found | https://github.com/getporter/porter/releases/download/cli/porter_parameters/ | porter parameters help [302 from https://getporter.org/cli/porter_parameters/] | https://getporter.org/introduction/intro-parameters/
404 Not Found | https://github.com/pulls?q=is:open+is:pr+user:getporter | /board/pull-requests [301 from https://getporter.org/board/pull-requests] | https://getporter.org/references/short-links/
404 Not Found | https://github.com/getporter/porter/releases/download/mixins-vs-plugins/ | Plugins are very different from mixins [302 from https://getporter.org/mixins-vs-plugins/] | https://getporter.org/introduction/intro-plugins/
404 Not Found | https://github.com/getporter/porter/releases/download/cli/porter_credentials/ | porter credentials [302 from https://getporter.org/cli/porter_credentials/] | https://getporter.org/references/cli/porter/
404 Not Found | https://github.com/getporter/porter/releases/download/cli/porter_installations/ | porter installations [302 from https://getporter.org/cli/porter_installations/] | https://getporter.org/references/cli/porter/
404 Not Found | https://github.com/getporter/porter/releases/download/cli/porter_mixins/ | porter mixins [302 from https://getporter.org/cli/porter_mixins/] | https://getporter.org/references/cli/porter/
404 Not Found | https://github.com/getporter/porter/releases/download/cli/porter_plugins/ | porter plugins [302 from https://getporter.org/cli/porter_plugins/] | https://getporter.org/references/cli/porter/
404 Not Found | https://github.com/getporter/porter/releases/download/cli/porter_bundles/ | porter bundles [302 from https://getporter.org/cli/porter_bundles/] | https://getporter.org/references/cli/porter/
404 Not Found | https://github.com/getporter/porter/releases/download/cli/porter/ | porter [302 from https://getporter.org/cli/porter/] | https://getporter.org/references/cli/bundles/
404 Not Found | https://docs.docker.com/docker-compose/reference/ | docker-compose CLI Command Reference | https://getporter.org/mixins/docker-compose/
404 Not Found | https://github.com/getporter/porter/releases/download/architecture-buildtime/bundle/custom-dockerfile/ | custom Dockerfile [302 from https://getporter.org/architecture-buildtime/bundle/custom-dockerfile/] | https://getporter.org/architecture-buildtime/
404 Not Found | https://github.com/getporter/porter/releases/download/mixin-dev-guide/architecture/ | Mixin Architecture [302 from https://getporter.org/mixin-dev-guide/architecture/] | https://getporter.org/architecture-buildtime/
404 Not Found | https://github.com/getporter/porter/releases/download/build-image/ | Building the Invocation Image [302 from https://getporter.org/build-image/] | https://getporter.org/architecture-buildtime/
404 Not Found | https://github.com/getporter/mixin-docker/ | docker mixin | https://getporter.org/blog/using-docker-in-bundles/
500 Internal Server Error | https://gist-it.appspot.com/https://github.com/getporter/porter/blob/main/examples/dockerapp/porter.yaml | script/src | https://getporter.org/blog/migrate-from-docker-app/
404 Not Found | https://github.com/getporter/porter/releases/download/end-users/installations/ | Managing installations [302 from https://getporter.org/end-users/installations/] | https://getporter.org/quickstart/desired-state/
404 Not Found | https://github.com/getporter/porter/releases/download/storage-migrate/ | https://getporter.org/storage-migrate [302 from https://getporter.org/storage-migrate] | https://getporter.org/cli/porter_storage_migrate/
404 Not Found | https://github.com/getporter/porter/releases/download/cli/porter_installation_apply/ | porter installation apply [302 from https://getporter.org/cli/porter_installation_apply/] | https://getporter.org/quickstart/desired-state/
404 Not Found | https://github.com/getporter/porter/releases/download/authors/templates/ | templates [302 from https://getporter.org/authors/templates/] | https://getporter.org/bundle/manifest/
404 Not Found | https://github.com/getporter/porter/releases/download/dependencies/ | dependencies [302 from https://getporter.org/dependencies] | https://getporter.org/wiring/
404 Not Found | https://github.com/getporter/porter/releases/download/copy-bundles/ | copied [302 from https://getporter.org/copy-bundles] | https://getporter.org/bundle/manifest/
404 Not Found | https://github.com/getporter/porter/releases/download/plugins/tutorial/ | plugins tutorial [302 from https://getporter.org/plugins/tutorial/] | https://getporter.org/blog/porter-collaboration/
404 Not Found | https://github.com/getporter/porter/releases/download/cli/porter_mixin_search/ | porter mixin search [302 from https://getporter.org/cli/porter_mixin_search/] | https://getporter.org/package-search/
404 Not Found | https://github.com/getporter/porter/releases/download/cli/porter_plugin_search/ | porter plugin search [302 from https://getporter.org/cli/porter_plugin_search/] | https://getporter.org/package-search/
404 Not Found | https://github.com/getporter/helm2-mixin/blob/main/pkg/helm/schema/schema.json | helm mixin schema [301 from https://getporter.org/helm-mixin/src/pkg/helm/schema/schema.json] [301 from https://github.com/getporter/helm-mixin/blob/main/pkg/helm/schema/schema.json] | https://getporter.org/mixin-dev-guide/commands/
404 Not Found | https://github.com/getporter/examples/blob/main/compose/ | full working example for how to use Docker Compose in a bundle [301 from https://getporter.org/examples/src/compose/] | https://getporter.org/blog/using-docker-in-bundles/
404 Not Found | https://github.com/getporter/porter/releases/download/mixin-dev-guide/distribution | https://getporter.org/mixin-dev-guide/distribution [302 from https://getporter.org/mixin-dev-guide/distribution] | https://getporter.org/cli/porter_mixins_feed_generate/

</body>
</html>

## Porter Command and Output
```
$ porter [e.g. install --param foo=bar --credential-set azure]
lots of output
```

## Version
Copy the output of `porter version` below

### PR DESCRIPTION
**Pull request by ludfjig**
# What does this change
Fixes 404s in docs.

# What issue does it fix
Closes #2866 

# Notes for the reviewer
_Put any questions or notes for the reviewer here._

# Checklist
- [ ] Did you write tests?
- [ ] Did you write documentation?
- [ ] Did you change porter.yaml or a storage document record? Update the corresponding schema file.
- [ ] If this is your first pull request, please add your name to the bottom of our [Contributors][contributors] list. Thank you for making Porter better! 🙇‍♀️

# Reviewer Checklist
* Comment with /azp run test-porter-release if a magefile or build script was modified
* Comment with /azp run porter-integration if it's a non-trivial PR

[contributors]: https://getporter.org/src/CONTRIBUTORS.md

### CODE REVIEW

File: docs/content/development/dist-a-mixin.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ systems:
 * GOOS: windows, linux, darwin
 * GOARCH: amd64
 
-If you are creating your mixin in Go, you may find the [Makefile][mk] that we use
+If you are creating your mixin in Go, you may find the [Makefile](TODO) that we use
```
Comment (ludfjig) [inline docs/content/development/dist-a-mixin.md:24]: I don't know what this is referencing, ideas?

File: docs/content/architecture-buildtime.md [hunk 2]
```diff
@@ -28,8 +28,8 @@ After the image is built, Porter then generates a CNAB-compliant bundle.json fil
 
 * [Create a bundle](/bundle/create/)
 * [Use a custom Dockerfile](/bundle/custom-dockerfile/)
-* [Mixin Architecture](/mixin-dev-guide/architecture/)
-* [Building the Invocation Image](/build-image/)
+* [Mixin Architecture](/how-to-guides/work-with-mixins)
+* [Building the Invocation Image](TODO)
```
Comment (ludfjig) [inline docs/content/architecture-buildtime.md:32]: not sure what page this is referring to

File: docs/content/references/cli/storage_migrate.md [hunk 3]
```diff
@@ -11,7 +11,7 @@ Migrate data from v0.38 to v1
 
 Migrate data from Porter v0.38 into a v1 installation of Porter.
 
-See https://getporter.org/storage-migrate for a full description of the migration process. Below is a summary:
+See todo for a full description of the migration process. Below is a summary:
```
Comment (ludfjig) [inline docs/content/references/cli/storage_migrate.md:14]: not sure where this is supposed to go

Comment (schristoff) [reply docs/content/references/cli/storage_migrate.md:14]: [Here](https://github.com/getporter/porter/blob/34ad0d1f95e6c53e27ddc20f77241692baac709a/docs/content/getting-started/migrate-v0-v1.md#L4)

File: docs/content/architecture-buildtime.md (hunk 2 above)
Comment (schristoff) [reply docs/content/architecture-buildtime.md:32]: [Unsure, just like here I guess?](https://github.com/getporter/porter/blob/34ad0d1f95e6c53e27ddc20f77241692baac709a/docs/content/how-to-guides/work-with-mixins.md)

Comment (ludfjig) [reply docs/content/architecture-buildtime.md:32]: I mean the second link

File: docs/content/references/cli/storage_migrate.md (hunk 3 above)
Comment (ludfjig) [reply docs/content/references/cli/storage_migrate.md:14]: thanks

File: docs/content/development/dist-a-mixin.md (hunk 1 above)
Comment (schristoff) [reply docs/content/development/dist-a-mixin.md:24]: Sub Makefile -> Magefile and so I would link them[ here I guess](https://github.com/getporter/skeletor/blob/main/magefile.go) 🤔

File: docs/content/architecture-buildtime.md (hunk 2 above)
Comment (schristoff) [reply docs/content/architecture-buildtime.md:32]: Just remove it for now I guess 🤷🏻 
If ya can

**Review by schristoff — APPROVED** [review]

**Pull request merged**